arwen/common.go

```diff
@@ -224,3 +227,37 @@ func (ac *AsyncGeneratedCall) GetValueBytes() []byte {
 func (ac *AsyncGeneratedCall) IsInterfaceNil() bool {
 	return ac == nil
 }
+
+func GetStorageLoadCost(trieDepth int64, metering MeteringContext, staticGasCost uint64, enableEpochsHandler vmcommon.EnableEpochsHandler) (uint64, error) {
+	if enableEpochsHandler.IsDynamicGasCostForDataTrieStorageLoadEnabled() {
+		return computeGasForStorageLoadBasedOnTrieDepth(trieDepth, metering.GasSchedule().DynamicStorageLoad)
+	}
+
+	return staticGasCost, nil
+}
+
+func computeGasForStorageLoadBasedOnTrieDepth(trieDepth int64, quadraticCoefficients config.DynamicStorageLoad) (uint64, error) {
```

Comment:
I would rename them simply `coefficients` as adding "quadratic" means that we use a quadratic equation, which might change in the future to something else.

Reply:
Renamed.